plugins/cindy3d/src/js/Triangles.js

```diff
@@ -173,8 +185,22 @@ Triangles.prototype.render = function(viewer) {
     gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER,
                      gl.LINEAR_MIPMAP_LINEAR);
     gl.hint(gl.GENERATE_MIPMAP_HINT, gl.NICEST);
-    gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE,
-                  this.texture.img);
+    let /** @type {HTMLImageElement|HTMLCanvasElement|HTMLVideoElement} */ img
+        = this.texture.img;
+    if (w !== this.texture.width || h !== this.texture.height) {
+      if (!this.textureScaler) {
+        this.textureScaler = /** @type {HTMLCanvasElement} */
+          (document.createElement("canvas"));
+        this.textureScaler.width = w;
+        this.textureScaler.height = h;
+      }
+      let ctx = this.textureScaler.getContext("2d");
+      ctx.clearRect(0, 0, w, h);
+      ctx.drawImage(this.texture.img, 0, 0, w, h);
```

Comment:
Scaling artifacts might occur with this approach. Probably one gets better results if we fill only one corner of img and modify the texture accesses (This has been done in CindyGL). But as long as we do not have an example, where these artifacts become visible, I guess it is okay.